torch/csrc/autograd/python_variable.cpp

```diff
@@ -151,10 +151,12 @@ static const char* VOLATILE_WARNING =
 
 static bool check_has_torch_dispatch(PyObject *obj) {
   PyTypeObject *tp = Py_TYPE(obj);
+  auto* attr = PyObject_FastGetAttrString(obj, "__torch_dispatch__").ptr();
```

Comment:
Not sure what the auto does here but this returns a new reference, so we actually own it.

Reply:
Ach, I see, the ptr() idiom is a bit dangerous but it works out because the PyObject in question has lifetime independent of PyObject_FastGetAttrString. (Usually.) Can fix anyway. There are a bunch of other sites that need to get similarly fixed too.

Reply:
> the PyObject in question has lifetime independent of PyObject_FastGetAttrString

What do you mean?
If this function has the same semantic as python's `PyObject_GetAttrString` then you get an owning reference that you have to decref here!

Reply:
But we return a py::object so explicit decref is not needed, the class handles it. But since it's a temporary in 154 I will immediately decref at the end of 154 even though I still have a raw pointer. In any case I have fixed it.